### ISSUE TITLE
Support for Haml 4.1.0 beta?

### ISSUE THREAD

**Issue by phansch** [brigade/haml-lint#228]
I am on `haml 4.1.0.beta.1` because of [cells-haml](https://github.com/trailblazer/cells-haml).
When I run haml-lint I get the following error:

```
$ haml-lint
Cannot handle Haml version: ~> 4.1
/home/<snip>/.rvm/gems/ruby-2.3.3/gems/haml_lint-0.24.0/lib/haml_lint/adapter.rb:21:in `detect_class'
/home/<snip>/.rvm/gems/ruby-2.3.3/gems/haml_lint-0.24.0/lib/haml_lint/document.rb:48:in `process_source'
/home/<snip>/.rvm/gems/ruby-2.3.3/gems/haml_lint-0.24.0/lib/haml_lint/document.rb:36:in `initialize'
/home/<snip>/.rvm/gems/ruby-2.3.3/gems/haml_lint-0.24.0/lib/haml_lint/runner.rb:77:in `new'
/home/<snip>/.rvm/gems/ruby-2.3.3/gems/haml_lint-0.24.0/lib/haml_lint/runner.rb:77:in `collect_lints'
/home/<snip>/.rvm/gems/ruby-2.3.3/gems/haml_lint-0.24.0/lib/haml_lint/runner.rb:119:in `process_file'
/home/<snip>/.rvm/gems/ruby-2.3.3/gems/haml_lint-0.24.0/lib/haml_lint/runner.rb:108:in `block in process_files'
/home/<snip>/.rvm/gems/ruby-2.3.3/gems/haml_lint-0.24.0/lib/haml_lint/runner.rb:107:in `each'
/home/<snip>/.rvm/gems/ruby-2.3.3/gems/haml_lint-0.24.0/lib/haml_lint/runner.rb:107:in `process_files'
/home/<snip>/.rvm/gems/ruby-2.3.3/gems/haml_lint-0.24.0/lib/haml_lint/runner.rb:132:in `report'
```
- Haml-Lint version: 0.24.0
- Haml version: 4.1.0.beta.1
- RuboCop version: 0.48.1
- Ruby version: 2.3.3

I know that haml 4.1 has been in beta for more than three years and it doesn't look like they will release a proper 4.1. It would still be nice if I could use haml-lint with this gem/version combination :sparkles:

**Comment by trotzig** [comment]
I don't have much context into this version, or why haml-lint chokes on it. Any help here would be appreciated! In the meantime, maybe there's a way you can run haml-lint on a different version of haml?

**Comment by phansch** [comment]
Unfortunately I'm currently stuck with haml 4.1.beta due to cells-haml and other libraries don't have support for haml 5, yet :/

I added 4.1 to `adapter.rb` and it seems to work without problems. However, the project I am testing this with, is not using any specific 4.1.beta features.

The changelog for haml 4.1 is [here](https://github.com/haml/haml/blob/4.1.0.beta.1/CHANGELOG.md)
The changeset between haml 4.0.5 and 4.1.beta.1 is [here](https://github.com/haml/haml/compare/4.0.5...4.1.0.beta.1)

If you want, I can open a PR that allows running with haml 4.1 and see where to go from there.

**Comment by trotzig** [comment]
That would be great, thanks!

**Comment by sds** [comment]
Implemented in #229. Thanks!

**Comment by phansch** [comment]
Thanks a lot!
